web-common/src/features/dashboards/state-managers/selectors/measures.ts

```diff
@@ -51,11 +52,27 @@ export const isMeasureValidPercentOfTotal = ({
   };
 };
 
-export const filterBasicMeasures = (
-  measures: MetricsViewSpecMeasureV2[] | undefined,
-) => measures?.filter((m) => !m.window) ?? [];
+export const filteredBasicMeasures = ({
+  metricsSpecQueryResult,
+  dashboard,
+}: DashboardDataSources) => {
+  return (includeComparisonMeasure: boolean) => {
+    includeComparisonMeasure =
+      includeComparisonMeasure &&
+      !!dashboard.showTimeComparison &&
+      !!dashboard.selectedComparisonTimeRange;
+    return (
+      metricsSpecQueryResult.data?.measures?.filter(
+        (m) =>
+          !m.window &&
+          (includeComparisonMeasure ||
+            m.type !== MetricsViewSpecMeasureType.MEASURE_TYPE_TIME_COMPARISON),
+      ) ?? []
+    );
+  };
+};
```

Comment:
**Issue**: The function 'filteredBasicMeasures' introduces a significant performance issue by potentially executing the filter operation every time it is invoked with a different 'includeComparisonMeasure' value. This can lead to unnecessary computations, especially in a reactive Svelte context where it might be re-evaluated frequently due to state changes. <br> **Fix**: To optimize performance, refactor the function to memoize the result based on the 'includeComparisonMeasure' and 'metricsSpecQueryResult' parameters. This can be achieved by using a memoization library or implementing a simple caching mechanism within the function. <br> **Code Suggestion**: 
 ```
 @@ -55,3 +73,10 @@
 +  const cache = new Map();
 +  return (includeComparison: boolean) => {
 +    const cacheKey = '${includeComparison}';
 +    if (cache.has(cacheKey)) {
 +      return cache.get(cacheKey);
 +    }
 +    const result = (metricsSpecQueryResult.data?.measures?.filter(
 +      (m) => !m.window && (includeComparison || m.type !== MetricsViewSpecMeasureType.MEASURE_TYPE_TIME_COMPARISON),
 +    ) ?? []);
 +    cache.set(cacheKey, result);
 +    return result;
 +  };
 ```

Reply:
The cost of converting `metricsSpecQueryResult` into a key for the cache is higher than iterating through the measures.